config/environments/development.rb

```diff
@@ -10,59 +9,72 @@
   # Do not eager load code on boot.
   config.eager_load = false
 
-  # Show full error reports and disable caching.
-  config.consider_all_requests_local       = true
-  config.action_controller.perform_caching = false
-
-  # action mailer config -- required for password resets and the bulk uploader
-    config.action_mailer.smtp_settings = {
-    :address   => "smtp.yourdomain.com",
-    :port      => 587, # ports 587 and 2525 are also supported with STARTTLS
-    :enable_starttls_auto => true, # detects and uses STARTTLS
-    :user_name => "XXXXX",
-    :password  => "XXXXX",
-    :authentication => 'login', 
-    :domain => 'yourdomain.com', # your domain to identify your server when connecting
-  }
-  config.action_mailer.default_url_options =  { host: 'localhost:3000' } #change this to match your server URL, i.e. www.fromthepage.com
-  config.action_mailer.default_options
+  # Show full error reports.
+  config.consider_all_requests_local = true
+
+  # Enable/disable caching. By default caching is disabled.
+  # Run rails dev:cache to toggle caching.
+  if Rails.root.join('tmp', 'caching-dev.txt').exist?
+    config.action_controller.perform_caching = true
+    config.action_controller.enable_fragment_cache_logging = true
+
+    config.cache_store = :memory_store
+    config.public_file_server.headers = {
+      'Cache-Control' => "public, max-age=#{2.days.to_i}"
+    }
+  else
+    config.action_controller.perform_caching = false
+
+    config.cache_store = :null_store
+  end
+
+  # Store uploaded files on the local file system (see config/storage.yml for options).
+  config.active_storage.service = :local
+
   # Don't care if the mailer can't send.
   config.action_mailer.raise_delivery_errors = false
 
-  ## Config for MailCatcher ##
-  # Install mailcatcher locally on your machine 'gem install mailcatcher'
-  # Run 'mailcatcher' in the terminal to start the server
-  # Open 'http://localhost:1080/' in your browser to see mail sent
-  config.action_mailer.delivery_method = :smtp
-  config.action_mailer.smtp_settings = { :address => "localhost", :port => 1025 }
+  config.action_mailer.perform_caching = false
```

Comment:
what does this do?

Reply:
@saracarl https://guides.rubyonrails.org/configuring.html#configuring-action-controller